integration-tests/src/tests/client/utils.rs

```diff
@@ -22,7 +23,9 @@ impl TestEnvNightshadeSetupExt for TestEnvBuilder {
             .into_iter()
             .zip(epoch_managers)
             .map(|(store, epoch_manager)| {
-                NightshadeRuntime::test(Path::new("../../../.."), store, genesis, epoch_manager)
+                let home_dir = Path::new("../../../..");
+                initialize_genesis_state(store.clone(), genesis, Some(home_dir));
```

Comment:
Hmm, it's not ideal to initialize genesis state here - since here we're initializing runtime and runtime should not need to initialize genesis state. I suppose this is also why you made KeyValueRuntime initialize genesis inside. Neither of these is ideal. We should initialize the genesis in the store before setting any of this stuff up.

But I also see that the problem is that we don't always have a genesis to test if we're using KeyValueRuntime. So this isn't so easy to refactor. Would you mind leaving a TODO here?

Reply:
Yep, exactly the problem I had hit. Ideally I understand this should go along with initializing store but we don't always do this as expected.

Further this is a consequence of the initialization moving out of the runtime that it's coupled with runtime here.